aeppl/math_stat.py

```diff
@@ -0,0 +1,62 @@
+import aesara
+import aesara.tensor as at
+from aesara.graph.basic import ancestors
+from aesara.graph.rewriting.basic import EquilibriumGraphRewriter, node_rewriter
+from aesara.scalar.basic import Add
+from aesara.tensor.elemwise import Elemwise
+from aesara.tensor.random.basic import NormalRV, normal
+from aesara.tensor.var import TensorConstant
+
+from aeppl.rewriting import logprob_rewrites_db
+
+
+@node_rewriter((Elemwise,))
+def add_independent_normals(fgraph, node):
+    if not isinstance(node.op.scalar_op, Add):
+        return None
+
+    X_rv, Y_rv = node.inputs
+
+    if not (X_rv.owner and Y_rv.owner) or not (
+        isinstance(X_rv.owner.op, NormalRV) and isinstance(Y_rv.owner.op, NormalRV)
+    ):
+        return None
+
+    # check independence
```

Comment:
I'm thinking more of the mathematical independence: $X \perp Y$. I am not even sure how this would manifest in Aesara/AePPL... We would have to use `MvNormalRV` instead?

Reply:
Afaiu two variables are (conditionally) independent if they have no common random variable ancestor in the graph. Did I miss something?